src/app/(main)/product/[id]/page.tsx

```diff
@@ -0,0 +1,97 @@
+"use client";
+
+import React from "react";
+import { use } from "react";
+import { useProduct } from "@cps/trpc/use-product";
+import Head from "next/head";
+import Button from "@cps/components/button";
+import ImageGallery from "@cps/components/image/image-gallery"; 
+
+interface ProductDetailsPageProps {
+  params: Promise<{ id: string }>; 
+}
+
+const ProductDetailsPage: React.FC<ProductDetailsPageProps> = ({ params }) => {
+  const { id } = use(params); // Unwrap Promise with `use`
+  const { data, error, isLoading } = useProduct(id);
+
+  if (isLoading) {
+    return (
+      <div className="flex justify-center items-center h-screen">
+        <p className="text-gray-500">Loading product details...</p>
+      </div>
+    );
+  }
+
+  if (error || !data) {
+    return (
+      <div className="flex justify-center items-center h-screen">
+        <p className="text-red-500">Error loading product details. Please try again later.</p>
+      </div>
+    );
+  }
+
+  const { name, description, images, prices } = data;
+
+  // Transform images data to match ImageGallery's expected format
+  const galleryImages = images.map((image) => ({
+    src: image.url,
+    alt: image.title || "Product image",
+  }));
+
+  return (
+    <div>
+      {/* Head for SEO */}
+      <Head>
+        <title>{name} - Product Details</title>
+        <meta name="description" content={description || ""} />
+      </Head>
```

Comment:
Instead of this, using https://nextjs.org/docs/app/api-reference/functions/generate-metadata is a better option.